dis_snek/models/discord/channel.py

```diff
@@ -1419,78 +1472,109 @@ async def archive(self, locked: bool = False, reason: Absent[str] = MISSING) ->
 
 @define()
 class GuildNewsThread(ThreadChannel):
-    async def edit(self, name, archived, auto_archive_duration, locked, rate_limit_per_user, reason) -> None:
+    async def edit(
+        self,
+        name: Absent[str] = MISSING,
+        archived: Absent[bool] = MISSING,
+        default_auto_archive_duration: Absent[AutoArchiveDuration] = MISSING,
+        locked: Absent[bool] = MISSING,
+        rate_limit_per_user: Absent[int] = MISSING,
+        reason: Absent[str] = MISSING,
+        **kwargs,
+    ) -> None:
         """
         Edit this thread.
 
         Args:
             name: 1-100 character channel name
             archived: whether the thread is archived
-            auto_archive_duration: duration in minutes to automatically archive the thread after recent activity, can be set to: 60, 1440, 4320, 10080
+            default_auto_archive_duration: duration in minutes to automatically archive the thread after recent activity, can be set to: 60, 1440, 4320, 10080
             locked: whether the thread is locked; when a thread is locked, only users with MANAGE_THREADS can unarchive it
             rate_limit_per_user: amount of seconds a user has to wait before sending another message (0-21600)
             reason: The reason for this change
         """
-        payload = {  # TODO Proper processing
-            "name": name,
-            "archived": archived,
-            "auto_archive_duration": auto_archive_duration,
-            "locked": locked,
-            "rate_limit_per_user": rate_limit_per_user,
-        }
-        await self._edit(payload=payload, reason=reason)
+        await super().edit(
+            name=name,
+            archived=archived,
+            default_auto_archive_duration=default_auto_archive_duration,
+            locked=locked,
+            rate_limit_per_user=rate_limit_per_user,
+            reason=reason,
+            **kwargs,
+        )
 
 
 @define()
 class GuildPublicThread(ThreadChannel):
-    async def edit(self, name, archived, auto_archive_duration, locked, rate_limit_per_user, reason) -> None:
+    async def edit(
+        self,
+        name: Absent[str] = MISSING,
+        archived: Absent[bool] = MISSING,
+        default_auto_archive_duration: Absent[AutoArchiveDuration] = MISSING,
+        locked: Absent[bool] = MISSING,
+        rate_limit_per_user: Absent[int] = MISSING,
+        reason: Absent[str] = MISSING,
+        **kwargs,
+    ) -> None:
         """
         Edit this thread.
 
         Args:
             name: 1-100 character channel name
             archived: whether the thread is archived
-            auto_archive_duration: duration in minutes to automatically archive the thread after recent activity, can be set to: 60, 1440, 4320, 10080
+            default_auto_archive_duration: duration in minutes to automatically archive the thread after recent activity, can be set to: 60, 1440, 4320, 10080
```

Comment:
Is there a reason you're renaming this?

Reply:
Bcu that is what the discord api named it, must have been a mistake from pre-alpha.

Yea... just noticed this will make it a breaking change... ohwells